frontend/views/pages/Contributions.vue

```diff
@@ -244,75 +274,90 @@ export default {
     openModal (modal) {
       sbp('okTurtles.events/emit', OPEN_MODAL, modal)
     },
-    async setPaymentInfo () {
-      if (this.$v.form.$invalid) {
-        alert(L('Invalid payment info'))
-        return
-      }
-      try {
-        const incomeDetailsType = this.form.incomeDetailsType
-        const groupProfileUpdate = await sbp('gi.contracts/group/groupProfileUpdate/create',
-          {
-            incomeDetailsType,
-            [incomeDetailsType]: +this.form[incomeDetailsType]
-          },
-          this.$store.state.currentGroupId
-        )
-        await sbp('backend/publishLogEntry', groupProfileUpdate)
-      } catch (e) {
-        console.error('setPaymentInfo', e)
-        alert(`Failed to update user's profile. Error: ${e.message}`)
-      }
-    },
-    toggleMenu () {
-      this.ephemeral.isActive = !this.ephemeral.isActive
-    },
-
-    textReceivingNonMonetary (contribution) {
-      // REVIEW - Is it safe to use v-html here? Related #502
-      return L('<strong>{what}</strong> from ', { what: contribution.what })
-    },
-    textReceivingMonetary (contribution) {
-      return L('<strong>Up to {amount} for mincome</strong> from ', {
-        amount: `${this.fakeStore.currency}${contribution}`
-      })
-    },
-
-    submitAddNonMonetary (value) {
-      console.log('TODO $store - submitAddNonMonetary')
-      this.fakeStore.giving.nonMonetary.push(value) // Hardcoded Solution
-    },
-    handleEditNonMonetary (value, index) {
-      if (!value) {
-        console.log('TODO $store - deleteNonMonetary')
-        this.fakeStore.giving.nonMonetary.splice(index, 1) // Hardcoded Solution
-      } else {
-        console.log('TODO $store - editNonMonetary')
-        this.$set(this.fakeStore.giving.nonMonetary, index, value) // Hardcoded Solution
-      }
-    },
-    handleFormTriggerClick () {
-      this.ephemeral.isEditingIncome = true
-    },
-    handleIncomeSave ({ canPledge, amount }) {
-      console.log('TODO $store - Save Income Details')
-      // -- Hardcoded Solution
-      this.fakeStore.receiving.monetary = canPledge ? null : this.fakeStore.mincome - amount
-      this.fakeStore.giving.monetary = canPledge ? amount : null
-      this.fakeStore.isFirstTime = false
-
-      this.closeIncome()
-    },
-    handleIncomeCancel () {
-      this.closeIncome()
+    async handleNonMonetary (type, value) {
+      const groupProfileUpdate = await sbp('gi.contracts/group/groupProfileUpdate/create',
+        { [type]: value },
+        this.$store.state.currentGroupId
+      )
+      await sbp('backend/publishLogEntry', groupProfileUpdate)
```

Comment:
It's important to handle exceptions around `async/await`, so if the update fails an error is displayed. In this case a simple + useful `catch (e)` would be:

```js
} catch (e) {
  console.error('handleNonMonetary', e)
  alert(e.message)
}
```